book/src/setup.md

```diff
@@ -0,0 +1,53 @@
+# Setup & installation
+
+This section provides an opinionated guide to setting up your development environment for working with Rust and Python.
+If you're an experienced developer in either language, feel free to skip this section.
+
+## Python
+
+For macOS/Linux users, it's recommended to manage Python versions using [pyenv](https://github.com/pyenv/pyenv). `pyenv` lets you easily switch between multiple versions of Python. It's simple, unobtrusive, and follows the UNIX tradition of single-purpose tools that do one thing well.
+
+Follow the instructions from the [installation steps](https://github.com/pyenv/pyenv?tab=readme-ov-file#installation) section of the README to install `pyenv` on your system.
+
+Windows users can use [pyenv-win](https://github.com/pyenv-win/pyenv-win), a fork of `pyenv` that allows you to install and manage Windows-native Python versions.
+
+### Python version
+
+This book uses Python 3.11.x. You can install the latest minor version using `pyenv`:
+
+```bash
+pyenv install 3.11.7
+```
+
+### Virtual environments
+
+It's recommended to use [virtual environments](https://docs.python.org/3/tutorial/venv.html) to manage your Python dependencies. This allows you to create isolated environments for each project, and avoid dependency conflicts between projects.
+
+The `venv` module is included in the Python standard library, so you don't need to install anything extra to use it.
+
+To create a virtual environment, run the following command:
+
+```bash
+# Setup a new environment for the first time
+python -m venv venv
+# Activate the environment
+source venv/bin/activate
```

Comment:
Since we mention Windows, the Windows command for this is different. I never can remember it so I'll have to do some digging to find it if we want to include it.

Reply:
Sure, I'll add a not above saying "To create a new virtual environment on Unix".